aws/resource_aws_dynamodb_table.go

```diff
@@ -816,15 +816,21 @@ func flattenAwsDynamoDbTableResource(d *schema.ResourceData, meta interface{}, t
 	if err != nil {
 		return err
 	}
-	timeToLive := []interface{}{}
-	attribute := map[string]*string{
-		"name": timeToLiveOutput.TimeToLiveDescription.AttributeName,
-		"type": timeToLiveOutput.TimeToLiveDescription.TimeToLiveStatus,
-	}
-	timeToLive = append(timeToLive, attribute)
-	d.Set("timeToLive", timeToLive)
 
-	log.Printf("[DEBUG] Loaded TimeToLive data for DynamoDB table '%s'", d.Id())
+	if timeToLiveOutput.TimeToLiveDescription != nil && timeToLiveOutput.TimeToLiveDescription.AttributeName != nil {
+		timeToLiveList := []interface{}{
```

Comment:
This is certainly more compact, but it's still a slice, not an array. The three dots I mentioned in my example were turning it into an array.

https://blog.golang.org/go-slices-usage-and-internals

It's certainly not a big deal though.